src/sentry/integrations/gitlab/webhooks.py

```diff
@@ -305,7 +305,7 @@ def post(self, request: Request) -> HttpResponse:
                 "reason"
             ] = "Gitlab's webhook secret does not match. Refresh token (or re-install the integration) by following this https://docs.sentry.io/product/integrations/integration-platform/public-integration/#refreshing-tokens."
             logger.exception(extra["reason"])
-            return HttpResponse(status=400, reason=extra["reason"])
+            return HttpResponse(status=409, reason=extra["reason"])
```

Comment:
Thank you for updating to match the right HTTP Response. `Bad Request` is undeniably, insufficient in conveying the right information.